### PR TITLE
conf on mirror seg doesn't get updated with mirror port number

### PR DESCRIPTION
**Pull request by shrakesh**
Updated gpcreateseg.sh to update the port after the copying data dir tomirror. pg_basebackup copies primary data to mirror.
Updated gpconfigurenewsegment to update the port on newly added mirrorsegment.
Updated gprecoverseg to update the port number after full recovery.
Added test case to validate mirror port in initstandby workflow.
Added test case to validate mirror port in gpexpand/gpaddmirror workflow.
Updated gprecoverseg unit test for port number update.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Review by piyushc01 — COMMENTED** [review]
Can we add a unit test for SED_PG_CONF function from gp_bash_functions.sh?
We have gp_bash_functions.feature file for the same.

**Comment by shrakesh** [comment]
> @piyushc01  ideally SED_PG_CONF should cover in unit test but currently we do not have any infra for unit test of bash function.  also there is plan to convert bash script to HLL so IMO no point to write a infra for bash function test.
As part of the pr code changes this already cover in gpinitsystem feature test where we are validating all the conf after gpinitsystem.

**Review by piyushc01 — APPROVED** [review]

File: gpMgmt/bin/lib/gpconfigurenewsegment [hunk 1]
```diff
@@ -175,6 +175,18 @@ class ConfExpSegCmd(Command):
...
+                    modifyConfCmd = ModifyConfSetting('Updating %s/postgresql.conf' % self.datadir,
+                                                      self.datadir + '/postgresql.conf',
+                                                      'port', self.port, optType='number')
+                    try:
+                        modifyConfCmd.run(validateAfter=True)
+                    except:
+                        self.set_results(modifyConfCmd.get_results())
+                        raise
+
                     return
```
Comment (pobbatihari) [inline gpMgmt/bin/lib/gpconfigurenewsegment:190]: Can we add unittests for these lines of code

File: gpMgmt/sbin/gpsegrecovery.py [hunk 2]
```diff
@@ -62,6 +63,17 @@ def run(self):
...
+        self.logger.info("Updating %s/postgresql.conf" % self.recovery_info.target_datadir)
+        modifyConfCmd = ModifyConfSetting('Updating %s/postgresql.conf' % self.recovery_info.target_datadir,
+                                          self.recovery_info.target_datadir + '/postgresql.conf',
+                                          'port', self.recovery_info.target_port, optType='number')
+        try:
+            modifyConfCmd.run(validateAfter=True)
+        except:
+            self.set_results(modifyConfCmd.get_results())
+            raise
+
```
Comment (pobbatihari) [inline gpMgmt/sbin/gpsegrecovery.py:76]: Can we add unittest cases for these lines of code

File: gpMgmt/test/behave/mgmt_utils/gpinitsystem.feature [hunk 3]
```diff
@@ -269,3 +269,13 @@ Feature: gpinitsystem tests
...
+    Scenario: gpinitsystem should create consistent port entry on segments postgresql.conf file
+        Given the database is not running
+        And the user runs command "rm -rf ../gpAux/gpdemo/datadirs/*"
+        And the user runs command "mkdir ../gpAux/gpdemo/datadirs/qddir; mkdir ../gpAux/gpdemo/datadirs/dbfast1; mkdir ../gpAux/gpdemo/datadirs/dbfast2; mkdir ../gpAux/gpdemo/datadirs/dbfast3"
+        And the user runs command "mkdir ../gpAux/gpdemo/datadirs/dbfast_mirror1; mkdir ../gpAux/gpdemo/datadirs/dbfast_mirror2; mkdir ../gpAux/gpdemo/datadirs/dbfast_mirror3"
+        And the user runs command "rm -rf /tmp/gpinitsystemtest && mkdir /tmp/gpinitsystemtest"
+        And the user runs "gpinitsystem -a -c ../gpAux/gpdemo/clusterConfigFile -l /tmp/gpinitsystemtest -P 21100 -h ../gpAux/gpdemo/hostfile"
+        And gpinitsystem should return a return code of 0
+        Then gpstate should return a return code of 0
+        And check segment conf: postgresql.conf
```
Comment (pobbatihari) [inline gpMgmt/test/behave/mgmt_utils/gpinitsystem.feature:281]: Can we improve the readability part of the testcase. Might be we have some steps to remove/create these directories, can we check on this

File: gpMgmt/sbin/gpsegrecovery.py (hunk 2 above)
Comment (shrakesh) [reply gpMgmt/sbin/gpsegrecovery.py:76]: this is covered under new changes in test_unit_gpsegrecovery.py and all this also covered in behave test. imo all path of this code changes is covered.

File: gpMgmt/bin/lib/gpconfigurenewsegment (hunk 1 above)
Comment (shrakesh) [reply gpMgmt/bin/lib/gpconfigurenewsegment:190]: ConfExpSegCmd is not covered under any unittest case, but the coverage is taken care by behave test. 
IMO we need to covered the complete class by unittest and that can be taken as different story/chore under code coverage.

**Review by pobbatihari — APPROVED** [review]
Looks good to me. Hope you have verified the pipeline of the branch

**Pull request merged**